spec/USERS.md

```diff
@@ -0,0 +1,72 @@
+
+## test_user
+`test_user` is the user created with a more limited set of privileges. It is created with the following
+roles:
+- readWrite on the ruby-driver database
+- dbAdmin on the ruby-driver database
+
+It is also granted the following roles against a database called "invalid_database." These permissions are used for the purpose of running tests against a database that doesn't exist.
+- readWrite on the invalid_database database
+- dbAdmin on the invalid_database database
+
+`test_user` also has the following roles, which are exclusively used to test transactions:
+- readWrite on the hr database
+- dbAdmin on the hr database
+- readWrite on the reporting database
+- dbAdmin on the reporting database
+
+The `test_user` has the username `test-user` and the password `password`; these values are not customizable without changing the source code.
+
+## User Creation
+
+Both users are typically created in the [spec_setup](support/spec_setup.rb) script, which can be
+run in two ways: either by running `bundle exec rake spec:prepare`, which only runs spec setup without
+running any actual tests, or by running `rake`, which runs spec setup and the entire test suite.
+
+First, the `spec_setup` script attempts to create the `root_user`. If this user already exists (for example,
+if you have already created this user in your test instance), `spec_setup` will skip this step. Once
+the script has verified the existence of `root_user`, it will create a client authenticated with the `root_user` and use that client to create a second user, `test_user`. Because `root_user` has the `userAdminAnyDatabase` role, it has the permissions necessary to create and destroy users on your MongoDB instance. If you have already created a user with the same credentials as `test_user` prior to running
+the `spec_setup` script, the script will delete this user and re-create it.
+
+The `root_user` is created in the `admin` database, while the `test_user` is created in the `ruby-driver`
+database.
+
+The authentication mechanism used to store the user credentials is going to change depending on the version of MongoDB running on your deployment. If you are running tests against a MongoDB instance with a server version older than 3.0, the users will be created using the `MONGODB-CR` authentication mechanism. If your server version is between 3.0 and 3.6 (inclusive), the test users will be created using the `SCRAM-SHA-1` mechanism, which was introduced as the new default starting in MongoDB version 3.0. If you are running a version of MongoDB newer than 4.0, test users will be authenticated using either `SCRAM-SHA-1` or `SCRAM-SHA-256`.
+
+**Note:** (m-launch)[http://blog.rueckstiess.com/mtools/mlaunch.html], the client tool we use to spin up MongoDB instances for our tests, creates users EXCLUSIVELY with the `SCRAM-SHA-1` mechanism, even when `SCRAM-SHA-256` is enabled on the test server. This should not impact your ability to run the Mongo Ruby Driver test suite.
+
```

Comment:
@p-mongo I went more into detail about the different versions of scram. I also wrote a note about what we discussed in #1540 in case somebody runs into similar issues that I had. Let me know what you think.